docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,254 @@
+        Target       time.Duration `json:"target,omitempty"`
+}
+```
+
+### Diagrams
+
+Much of this is better explained visually:
+
+Below is a detailed view of the BackendPool
+
+![BackendPool](./images/bep.svg)
+
+This diagram lightly follows the example request for a model `interestingName`. 
+The flow can be described as:
+- The request comes in to our routing solution(Ext-Proc)
+- ExtProc looks up the UseCases affiliated with this pool `examplePool`
+- `interestingName` is currently undergoing a change of LoRA adapters from `creativeNameGen-v3` (20% traffic split) to `veryCreativeNameGen` (80% traffic split)
+- `veryCreativeNameGen` is selected as the LoRA adapter, and replaces `interestingName` in the body of the request (mutated by ext-proc) 
+- the request is then efficiently scheduled onto one of the valid Pods
+- metrics are sent back to the BEP, aggregated and re-emitted via sidecar (following the metric standardization)
+
+How Multiple BackendPools might integrate together:
+
+![K8s Gateway with BackendPools](./images/gw_w_bep.svg)
+
+Here we see that we can have:
+- Multiple Routes pointing to the same pool
+- Routes splitting traffic across multiple pools
+
+The functionality of the Kubernetes Gateway is unchanged with this proposal, allowing seamless integration with the BackendPool.
+
+
+### Alternatives
+
+#### Key Decisions
+
+Our alternatives hinge on some key decisions:
+- Allowing HTTPRoute to treat the BackendPool as the backendRef
+- Creating a separate layer of abstraction, instead of extending HTTPRoute
+
+#### LLMUseCase as a backend ref
+
+We toyed with the idea of allowing an LLMUsecase be the target of an HTTPRouteRules backend ref. However, doing so would require the Kubernetes Gateway to be able to interpret body level parameters, and require that the HTTPRoute also specify the backend the UseCase is intended to run on. All of which would require substantial work on the Kubernetes Gateway, while not providing much flexibility.
+
+#### LLMRoute
+
+Our original idea was to define all UseCase config at the Kubernetes Gateway layer, and have no BackendPool. This is inherently challenging, as LLMRoute would become a superset of HTTPRoute, or the Gateway would become bespoke, and work only for the LLMRoute use case.
+
+## FAQ
+- Why 2 layers of weighting? (HttpRoute & UseCase)
+  - Feasibly done - No extension of HttpRoute. Just works, as BackendPool operates like a service.
+  - Complexity is only expressed during transition states (model version upgrade)
+  - Keeps Pools self contained - multiple K8s gateways can direct traffic to the same pool without needing to re-express Pool-level behavior
+- What is a backend pool attempting to define?
+  - BackendPool groups resources that should be shared over the UseCases that are affiliated with the pool
+  - Best practice would also suggest keeping the same base model for all ModelServers in the pool, but that is not enforced
+
+
+## Open Questions
+
+- Reasonable defaults
```

Comment:
Should this be done through serving catalog? Or examples should provide good starting configs

Reply:
I think it could be interesting to have presets available in the serving catalog. That would definitely help with adoption.

Reply:
I really like the idea of having examples within serving catalog also.

This bullet was meant more as a discussion point around how we ensure unsurprising behavior when a user creates a BackendPool/LLMUseCase when they do not declare optional fields. However I do see how the interpretation was taken this way. I tried to add a tad more clarification. LMK if that's helpful at all.